MyStructures/MyStructures/Class1.cs

```diff
@@ -1,5 +1,119 @@
-﻿namespace MyStructures;
+﻿using System.Security.AccessControl;
 
-public class Class1
+namespace Test
 {
-}
\ No newline at end of file
+  /// <summary>
+  /// Список
+  /// </summary>
+  public class MyList
+  {
+    private int[] _array;
+    private int _cursor;
+    private MyList _rangeArray;
```

Comment:
```suggestion
    private MyList _rangeArray;
```

Не стоит выносить это в отдельное поле, оставь локальной переменной внутри метода
ВЫНОСИ в отдельные поля  только те обьекты, которые инициализируются в конструкторе, иначе может быть NullReferenceException